File: 20. Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """

        stack = []

        for i in s:
            if(i=="["):
                stack.append("]")
            elif(i=="{"):
                stack.append("}")
            elif(i=="("):
                stack.append(")")
            else:
                if stack:
                    if(stack.pop()!=i):
                        return False
                else:
                    return False
        return not any(stack)

File: 20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) 
    {
        stack<char> code;

        for(int i = 0;i<s.size();i++)
        {
            switch(s[i])
            {
                case '(':
                    code.push(')'); //Adding antimatter so i can neutralize when there occurs matter
                    break;

                case '[':
                    code.push(']'); //Adding antimatter so i can neutralize when there occurs matter
                    break;

                case '{':
                    code.push('}'); //Adding antimatter so i can neutralize when there occurs matter
                    break;
                default:
                    if(code.empty())
                    {
                        return false;
                    }
                    char top = code.top();
                    if(top==s[i])
                    {
                        code.pop();
                    }
                    else
                    {
                        return false;
                    }
            }
        }
        if(!code.empty())
        {
            return false;
        }
        return true;
    }
};